homeassistant/components/onewire/sensor.py

```diff
@@ -400,6 +409,28 @@ def get_entities(
                         description.device_class = SensorDeviceClass.HUMIDITY
                         description.native_unit_of_measurement = PERCENTAGE
                         description.name = f"Wetness {s_id}"
+                _LOGGER.info(
+                    "\n---- GET_ENTITIES ----\n -- Description: %s",
+                    description,
+                )
+                if (
+                    SENSOR_PRECISION_CONFIG_OPTION in options
+                    and device.id in options[SENSOR_PRECISION_CONFIG_OPTION]
+                ):
+                    sensor_precision_config = options[SENSOR_PRECISION_CONFIG_OPTION][
+                        device.id
+                    ]
+                    if sensor_precision_config in PRECISION_MAPPING_DS18B20:
+                        description = copy.deepcopy(description)
+                        description.override_key = PRECISION_MAPPING_DS18B20[
```

Comment:
I think you should convert `override_key` to a lambda: `Callable[[ConfigEntry], str]`
`lambda e: "k"`, etc

Reply:
Currently, you have:
```python
class OneWireSensorEntityDescription(OneWireEntityDescription, SensorEntityDescription):
    """Class describing OneWire sensor entities."""
    override_key: str | None = None
```

You should replace it with:
```python
class OneWireSensorEntityDescription(OneWireEntityDescription, SensorEntityDescription):
    """Class describing OneWire sensor entities."""
    override_key: Callable[[ConfigEntry, str], str] | None = None
```

Reply:
Then, in the entity_description, you can replace:
```python
    override_key="typeK/temperature",
```

with:
```python
    override_key=lambda e, d: "typeK/temperature",
```

Reply:
And for the DS18B20 you set it to:
```python
    override_key=_get_ds18b20_precision,
```
with
```python
def _get_ds18b20_precision(entry: ConfigEntry, device_id: str) -> str:
    ...
```

Reply:
As a final step, you use this for the device_file:
```python
    key = description.key
    if description.override_key:
        key = description.override_key(entry, device_id) or description.key
    device_file = os.path.join(..., key )
```

Reply:
I have create a PR against your fork: https://github.com/droberg/core/pull/2
Feel free to either merge it, or use it simply for inspiration.